crates/polars-core/src/chunked_array/logical/datetime.rs

```diff
@@ -81,21 +81,39 @@ impl LogicalType for DatetimeChunked {
             },
             #[cfg(feature = "dtype-time")]
             (Datetime(tu, _), Time) => match tu {
-                TimeUnit::Nanoseconds => Ok((self.0.as_ref() % NS_IN_DAY)
```

Comment:
the issue with the current implementation is that `self.0.as_ref() % NS_IN_DAY` might return a negative value